<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SMEditorWindow</class>
 <widget class="QMainWindow" name="SMEditorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Cyberiada State Machine Editor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="smeditor.qrc">
    <normaloff>:/Icons/images/cyberiada.png</normaloff>:/Icons/images/cyberiada.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QSplitter" name="vSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QSplitter" name="hSplitter">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <widget class="CyberiadaSMView" name="SMView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
        <property name="animated">
         <bool>true</bool>
        </property>
        <attribute name="headerVisible">
         <bool>false</bool>
        </attribute>
       </widget>
       <widget class="CyberiadaSMPropertiesWidget" name="propertiesWidget" native="true"/>
      </widget>
      <widget class="CyberiadaSMGraphicsView" name="sceneView"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="actionExport"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionFont"/>
    <addaction name="separator"/>
    <addaction name="actionZoomIn"/>
    <addaction name="actionZoomOut"/>
    <addaction name="actionFitContent"/>
    <addaction name="separator"/>
    <addaction name="actionTransitionText"/>
    <addaction name="actionInspectorMode"/>
    <addaction name="actionGridVisibility"/>
    <addaction name="actionSnapMode"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuElements">
    <property name="title">
     <string>Elements</string>
    </property>
    <addaction name="actionNewState"/>
    <addaction name="actionNewInitial"/>
    <addaction name="actionNewFinal"/>
    <addaction name="actionNewTerminate"/>
    <addaction name="actionNewComment"/>
    <addaction name="actionNewFormalComment"/>
    <addaction name="actionNewTransition"/>
    <addaction name="actionNewChoise"/>
    <addaction name="actionNewStateMachine"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuElements"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionZoomIn"/>
   <addaction name="actionZoomOut"/>
   <addaction name="actionFitContent"/>
   <addaction name="separator"/>
   <addaction name="actionSelectTool"/>
   <addaction name="actionPan"/>
   <addaction name="separator"/>
   <addaction name="actionGridVisibility"/>
   <addaction name="actionInspectorMode"/>
   <addaction name="actionPolylineMode"/>
  </widget>
  <widget class="QToolBar" name="elementToolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNewStateMachine"/>
   <addaction name="actionNewState"/>
   <addaction name="actionNewInitial"/>
   <addaction name="actionNewFinal"/>
   <addaction name="actionNewTerminate"/>
   <addaction name="actionNewComment"/>
   <addaction name="actionNewFormalComment"/>
   <addaction name="actionNewTransition"/>
   <addaction name="separator"/>
   <addaction name="actionDeleteElement"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionFont">
   <property name="text">
    <string>Font</string>
   </property>
  </action>
  <action name="actionZoomIn">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/zoom-in.png</normaloff>:/Icons/images/zoom-in.png</iconset>
   </property>
   <property name="text">
    <string>Zoom In</string>
   </property>
  </action>
  <action name="actionZoomOut">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/zoom-out.png</normaloff>:/Icons/images/zoom-out.png</iconset>
   </property>
   <property name="text">
    <string>Zoom Out</string>
   </property>
  </action>
  <action name="actionFitContent">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/fit-content.png</normaloff>:/Icons/images/fit-content.png</iconset>
   </property>
   <property name="text">
    <string>Fit Content</string>
   </property>
  </action>
  <action name="actionPan">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/hand.png</normaloff>:/Icons/images/hand.png</iconset>
   </property>
   <property name="text">
    <string>Pan</string>
   </property>
   <property name="toolTip">
    <string>Pan</string>
   </property>
  </action>
  <action name="actionSelectTool">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/cursor.png</normaloff>:/Icons/images/cursor.png</iconset>
   </property>
   <property name="text">
    <string>Select Tool</string>
   </property>
   <property name="toolTip">
    <string>Select Tool</string>
   </property>
  </action>
  <action name="actionTransitionText">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Текст на переходах</string>
   </property>
  </action>
  <action name="actionInspectorMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/inspector-mode.png</normaloff>:/Icons/images/inspector-mode.png</iconset>
   </property>
   <property name="text">
    <string>Режим инспектора</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Настройки</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actionDeleteElement">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/remove.png</normaloff>:/Icons/images/remove.png</iconset>
   </property>
   <property name="text">
    <string>Delete Element</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>&amp;Save as...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionSnapMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/snap-mode.png</normaloff>:/Icons/images/snap-mode.png</iconset>
   </property>
   <property name="text">
    <string>Прилипание</string>
   </property>
  </action>
  <action name="actionNewState">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/state.png</normaloff>:/Icons/images/state.png</iconset>
   </property>
   <property name="text">
    <string>New State</string>
   </property>
  </action>
  <action name="actionNewInitial">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/init-state.png</normaloff>:/Icons/images/init-state.png</iconset>
   </property>
   <property name="text">
    <string>New Initial</string>
   </property>
  </action>
  <action name="actionNewFinal">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/final-state.png</normaloff>:/Icons/images/final-state.png</iconset>
   </property>
   <property name="text">
    <string>New Final</string>
   </property>
   <property name="toolTip">
    <string>New Final</string>
   </property>
  </action>
  <action name="actionNewTerminate">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/terminate.png</normaloff>:/Icons/images/terminate.png</iconset>
   </property>
   <property name="text">
    <string>New Terminate</string>
   </property>
  </action>
  <action name="actionNewFormalComment">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/comment-machine.png</normaloff>:/Icons/images/comment-machine.png</iconset>
   </property>
   <property name="text">
    <string>New Formal Comment</string>
   </property>
  </action>
  <action name="actionNewComment">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/comment.png</normaloff>:/Icons/images/comment.png</iconset>
   </property>
   <property name="text">
    <string>New Comment</string>
   </property>
  </action>
  <action name="actionNewTransition">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/trans.png</normaloff>:/Icons/images/trans.png</iconset>
   </property>
   <property name="text">
    <string>New Transition</string>
   </property>
  </action>
  <action name="actionNewChoise">
   <property name="text">
    <string>New Choise</string>
   </property>
  </action>
  <action name="actionNewStateMachine">
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/sm.png</normaloff>:/Icons/images/sm.png</iconset>
   </property>
   <property name="text">
    <string>New State Machine</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="text">
    <string>Export...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>New File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionGridVisibility">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/grid.png</normaloff>:/Icons/images/grid.png</iconset>
   </property>
   <property name="text">
    <string>Grid</string>
   </property>
  </action>
  <action name="actionPolylineMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="smeditor.qrc">
     <normaloff>:/Icons/images/polyline-mode.png</normaloff>:/Icons/images/polyline-mode.png</iconset>
   </property>
   <property name="text">
    <string>Режим полилинии</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CyberiadaSMView</class>
   <extends>QTreeView</extends>
   <header>cyberiadasm_view.h</header>
   <slots>
    <signal>currentIndexActivated(QModelIndex)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>CyberiadaSMPropertiesWidget</class>
   <extends>QWidget</extends>
   <header>cyberiadasm_properties_widget.h</header>
   <container>1</container>
   <slots>
    <slot>slotElementSelected(QModelIndex)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>CyberiadaSMGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>cyberiadasm_editor_view.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="smeditor.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotFileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SMView</sender>
   <signal>currentIndexActivated(QModelIndex)</signal>
   <receiver>propertiesWidget</receiver>
   <slot>slotElementSelected(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>509</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>509</x>
     <y>767</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotFileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveAs</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotFileSaveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionInspectorMode</sender>
   <signal>triggered(bool)</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotInspectorModeTriggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSnapMode</sender>
   <signal>triggered(bool)</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotSnapModeTriggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTransitionText</sender>
   <signal>triggered(bool)</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotShowTransitionActionTriggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewStateMachine</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotNewSM()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewState</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotNewState()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewInitial</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotNewInitial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewFinal</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotNewFinal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewTerminate</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotNewTerminate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewChoise</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotNewChoise()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewComment</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotNewComment()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewFormalComment</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotNewFormalComment()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewTransition</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotNewTransition()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExport</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotFileExport()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionGridVisibility</sender>
   <signal>triggered(bool)</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotGridVisibilityTriggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPreferences</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotPreferences()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDeleteElement</sender>
   <signal>triggered()</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotDeleteElement()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPolylineMode</sender>
   <signal>triggered(bool)</signal>
   <receiver>SMEditorWindow</receiver>
   <slot>slotPolylineModeTriggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slotFileOpen()</slot>
  <slot>slotFileSave()</slot>
  <slot>slotFileSaveAs()</slot>
  <slot>slotInspectorModeTriggered(bool)</slot>
  <slot>slotSnapModeTriggered(bool)</slot>
  <slot>slotShowTransitionActionTriggered(bool)</slot>
  <slot>slotNewSM()</slot>
  <slot>slotNewState()</slot>
  <slot>slotNewInitial()</slot>
  <slot>slotNewFinal()</slot>
  <slot>slotNewTerminate()</slot>
  <slot>slotNewComment()</slot>
  <slot>slotNewFormalComment()</slot>
  <slot>slotNewTransition()</slot>
  <slot>slotNewChoise()</slot>
  <slot>slotFileExport()</slot>
  <slot>slotGridVisibilityTriggered(bool)</slot>
  <slot>slotPreferences()</slot>
  <slot>slotDeleteElement()</slot>
  <slot>slotPolylineModeTriggered(bool)</slot>
 </slots>
</ui>
